frontend/projects/upgrade/src/app/core/segments/segments.service.ts

```diff
@@ -9,21 +9,43 @@ import {
   selectExperimentSegmentsInclusion,
   selectExperimentSegmentsExclusion,
   selectSegmentById,
+  selectSearchString,
+  selectSearchKey,
+  selectSortKey,
+  selectSortAs,
 } from './store/segments.selectors';
-import { SegmentInput, UpsertSegmentType } from './store/segments.model';
-import { filter, map, tap } from 'rxjs/operators';
+import { SegmentInput, SegmentLocalStorageKeys, UpsertSegmentType } from './store/segments.model';
+import { filter, first, map, tap } from 'rxjs/operators';
 import { Observable, combineLatest } from 'rxjs';
 import { SegmentsDataService } from './segments.data.service';
+import { SEGMENT_SEARCH_KEY, SEGMENT_SORT_AS, SEGMENT_SORT_KEY } from '../../../../../../../types/src/Experiment/enums';
```

Comment:
Is there a reason we're not importing these from 'upgrade_types', like we do for experiments?